### PR TITLE
change(deps): Upgrade tracing-subscriber and related dependencies

### LINKED ISSUE #2953: Upgrade tracing to the next major version
## Motivation

`tracing` has a medium-priority security fix:
* advisory: https://rustsec.org/advisories/RUSTSEC-2020-0159
* original issue: https://github.com/chronotope/chrono/issues/499

(Zebra uses UTC, so our direct uses of `chrono` are safe.)

`tracing-error` has a low-priority security fix, because it doesn't actually use the vulnerable API in `thread_local` 1.0:
* advisory: https://rustsec.org/advisories/RUSTSEC-2022-0006.html
* API analysis: https://github.com/tokio-rs/tracing/pull/1858#pullrequestreview-861344519

## Tasks

1. All these tracing crates need to be upgraded together:

Core tracing:
`tracing-subscriber`: https://github.com/ZcashFoundation/zebra/pull/2956
`tracing-error`: https://github.com/ZcashFoundation/zebra/pull/2948

Tracing features:
`tracing-journald`: https://github.com/ZcashFoundation/zebra/pull/2949
`tracing-flame`: https://github.com/ZcashFoundation/zebra/pull/3265
`sentry-tracing`: https://github.com/ZcashFoundation/zebra/pull/3381
`sentry`: https://github.com/ZcashFoundation/zebra/pull/3382
`color-eyre`: https://github.com/ZcashFoundation/zebra/pull/3342, #3343

Optional framework upgrades:
- `abscissa`: `zebrad` has its own `tracing` component, so upgrading `abscissa` is optional
- `gumdrop`: does not depend on `tracing`, should be replaced with `clap` in `abscissa 0.6`

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

Before we integrate tokio-console (#2112), we need to upgrade tracing-subscriber to the latest version.

Close #2953.

## Solution

Upgrade dependencies using `cargo upgrade`:
* tracing-error
* tracing-subscrber
    * Enable the env-filter feature
* color-eyre
    * Disable span traces and track caller by default, to improve performance
* tracing-flame
* tracing-journald
* sentry
* sentry-tracing
* metrics
    * Fix type inference for metrics
* metrics-exporter-prometheus
    * Update metrics-exporter-prometheus API 
    * Disable unused features
* reqwest

Update duplicate dependency checks
Tidy up some unrelated test code

## Review

Anyone can review this PR.

I've marked it as a high priority because it could cause merge conflicts with other PRs.

### Reviewer Checklist

  - [ ] Existing tests pass

### CODE REVIEW

**Comment by teor2345** [comment]
These upgrades could cause sync speed changes, so I'm running a full sync here:
https://github.com/ZcashFoundation/zebra/actions/runs/2395236167

The current time is `5h50m`.

Edit: I had to restart the full sync, because the test regexes needed fixes for the new log format.

File: zebrad/tests/acceptance.rs
Comment (conradoplg) [inline zebrad/tests/acceptance.rs]:
```suggestion
        zebrad.expect_stdout_line_matches("loaded Zebra state cache .*tip.*=.*None")?;
```

File: zebrad/tests/acceptance.rs
Comment (conradoplg) [inline zebrad/tests/acceptance.rs]:
```suggestion
        zebrad.expect_stdout_line_matches(".*sync_percent.*=.*100")?;
```

File: zebrad/tests/acceptance.rs
Comment (conradoplg) [inline zebrad/tests/acceptance.rs]:
```suggestion
        zebrad.expect_stdout_line_matches(".*sync_percent.*=.*100")?;
```

File: zebrad/tests/common/lightwalletd.rs
Comment (conradoplg) [inline zebrad/tests/common/lightwalletd.rs]:
```suggestion
            zebrad_failure_messages.push("loaded Zebra state cache .*tip.*=.*None".to_string());
```

File: zebrad/tests/common/sync.rs
Comment (conradoplg) [inline zebrad/tests/common/sync.rs]:
```suggestion
    "finished initial sync to chain tip, using gossiped blocks .*sync_percent.*=.*100";
```

File: zebrad/tests/common/sync.rs
Comment (conradoplg) [inline zebrad/tests/common/sync.rs]:
```suggestion
    "estimated progress to chain tip .*sync_percent.*=.*97";
```

**Review by conradoplg — CHANGES_REQUESTED** [review]
Seeing the raw logs, it seems the regexes need further adjustments.

Also, would it better/possible to log without the fancy stuffy to make this easier?

**Comment by teor2345** [comment]
I think I've fixed the log matching issues now, we'll find out when the test runs are finished.

> Also, would it better/possible to log without the fancy stuffy to make this easier?

I'm hoping these regex changes will work, regardless of the exact log format.

We can turn off colour, but that makes the logs less readable, and the tests will break as soon as someone turns it on:
https://github.com/ZcashFoundation/zebra/blob/9a8ab9468df14a56aa80392bb3be654ec00cbd0e/zebrad/tests/common/config.rs#L49-L56

There is also a `Compact` format, but it's still coloured, so it still has these issues:
https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/format/index.html

**Comment by teor2345** [comment]
This appears to work now, the macOS CI failure seems to be a parameter download or cache issue.

**Comment by teor2345** [comment]
I started another full sync here, to check that performance is the same or better.

It should take around 5h35m for the full sync job:
https://github.com/ZcashFoundation/zebra/actions/runs/2406322345

**Comment by teor2345** [comment]
That didn't quite work, the regexes were too permissive this time around.
(So the full sync "finished" in 6 minutes.)

This run should take around `5h35m`:
https://github.com/ZcashFoundation/zebra/actions/runs/2408014484

**Comment by conradoplg** [comment]
> That didn't quite work, the regexes were too permissive this time around. (So the full sync "finished" in 6 minutes.)

Oops :grimacing: 
 
> This run should take around `5h35m`: https://github.com/ZcashFoundation/zebra/actions/runs/2408014484

It seems the full sync test hasn't run, and I can't understand why

**Comment by teor2345** [comment]
> It seems the full sync test hasn't run, and I can't understand why

Let's try again, but with the right parameters this time:
https://github.com/ZcashFoundation/zebra/actions/runs/2411388484

**Comment by teor2345** [comment]
These upgrades slow down sync by about 15%, so we can't merge it yet.

**Comment by teor2345** [comment]
I disabled some color-eyre features by default, let's hope that's faster:
https://github.com/ZcashFoundation/zebra/actions/runs/2412599719

**Comment by conradoplg** [comment]
5h 55m, which I guess is not ideal yet :cry:

File: zebrad/tests/common/launch.rs
Comment (conradoplg) [inline zebrad/tests/common/launch.rs:48]: Was this intended?

Comment (teor2345) [reply zebrad/tests/common/launch.rs:48]: Some tests were sometimes failing due to sync timeouts, this will be fixed in PR #4474.

**Comment by teor2345** [comment]
> 5h 55m, which I guess is not ideal yet 😢

It goes down to 5h43m after the tokio-console refactor in PR #4519, which I think is acceptable:
https://github.com/ZcashFoundation/zebra/runs/6666292148

Once that's merged, we can find some long-term fixes using tokio-console.

**Review by conradoplg — APPROVED** [review]

**Comment by teor2345** [comment]
I'm merging this PR manually, CI is broken due to bug #4551 in lightwalletd.

**Pull request merged**
